File: backend/ModuMessenger/src/main/java/com/example/modumessenger/common/exception/CustomException.java
package com.example.modumessenger.common.exception;

import lombok.Getter;

import java.util.Set;

@Getter
public class CustomException extends RuntimeException {

    private ErrorCode errorCode;
    private String errorTarget;

    public CustomException(ErrorCode errorCode) {
        this.errorCode = errorCode;
        this.errorTarget = "";
    }

    public CustomException(ErrorCode errorCode, String errorTarget) {
        this.errorCode = errorCode;
        this.errorTarget = errorTarget;
    }

    public CustomException(ErrorCode errorCode, Long id) {
        this.errorCode = errorCode;
        this.errorTarget = String.valueOf(id);
    }

    public CustomException(ErrorCode errorCode, Set<Long> ids) {
        this.errorCode = errorCode;
        this.errorTarget = setToString(ids);
    }

    private String setToString(Set<Long> ids) {
        StringBuilder sb = new StringBuilder();
        for(Long id : ids) {
            sb.append(id).append(",");
        }
        sb.deleteCharAt(sb.length()-1);
        return sb.toString();
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/global/exception/CustomException.kt
package com.example.memberservice.global.exception

class CustomException : RuntimeException {

    val errorCode: ErrorCode
    val errorTarget: String

    constructor(errorCode: ErrorCode) {
        this.errorCode = errorCode
        this.errorTarget = ""
    }

    constructor(errorCode: ErrorCode, errorTarget: String) {
        this.errorCode = errorCode
        this.errorTarget = errorTarget
    }

    constructor(errorCode: ErrorCode, id: Long?) {
        this.errorCode = errorCode
        this.errorTarget = id.toString()
    }

    constructor(errorCode: ErrorCode, ids: Set<Long>) {
        this.errorCode = errorCode
        this.errorTarget = ids.joinToString(",")
    }
}
